shared-module/displayio/Display.c

```diff
@@ -240,11 +243,23 @@ STATIC bool _refresh_area(displayio_display_obj_t* self, const displayio_area_t*
     if (!displayio_display_core_clip_area(&self->core, area, &clipped)) {
         return true;
     }
-    uint16_t subrectangles = 1;
     uint16_t rows_per_buffer = displayio_area_height(&clipped);
     uint8_t pixels_per_word = (sizeof(uint32_t) * 8) / self->core.colorspace.depth;
     uint16_t pixels_per_buffer = displayio_area_size(&clipped);
-    if (displayio_area_size(&clipped) > buffer_size * pixels_per_word) {
+    
+    uint16_t subrectangles = 1;
+    // for SH1107 and other boundary constrained controllers
+    //      write one single row at a time
+    if (self->SH1107_addressing) {
+        subrectangles = rows_per_buffer;  // vertical (column mode) write each separately (height times)
+    }
+
+    // Skip this recalculation of subrectangles if SH1107 (each subrectangle is a single row)
+    // [mdroberts1243] I am worried/confused about the pixels_in_byte_share_row calculation though
+    //       since it makes sense to be on a byte boundary (actually a page boundary too)
+    //       seems to work as is though.
```

Comment:
I think it's fine to rely on your testing. pixels_in_byte_share_row should be what you want.

Reply:
O.k. I might leave the comment in place though.  It seems the byte boundary is taken care of by the pixels_in_byte_share_row setting elsewhere, but the fact that this section claims to align on byte boundaries bothers me, because I'm skipping it!